ruby/ql/lib/codeql/ruby/frameworks/ActionView.qll

```diff
@@ -216,4 +216,138 @@ class FileSystemResolverAccess extends DataFlow::CallNode, FileSystemAccess::Ran
 
   override DataFlow::Node getAPathArgument() { result = this.getArgument(0) }
 }
+
 // TODO: model flow in/out of template files properly,
+// TODO: Move the classes and predicates above inside this module.
+/** Modeling for `ActionView`. */
+module ActionView {
+  /**
+   * Action view helper methods which are XSS sinks.
+   */
+  module Helpers {
+    /**
+     * A call to an ActionView helper which renders its argument without escaping.
+     * The argument should be treated as an XSS sink. In the documentation for
+     * classes in this module, the vulnerable argument is named `x`.
+     */
+    abstract class RawHelperCall extends MethodCall {
```

Comment:
We should avoid introducing public abstract classes when we don't want/expect users to provide their own implementations. However, I'm already working on a branch to remove the other abstract classes in our framework libraries, so I suggest merging this PR as-is, and then I can fix this one as well.